api/query/cache/poll/poll.go

```diff
@@ -84,20 +87,77 @@ func wait(start time.Time, total time.Duration) {
 	}
 }
 
-// KeepMetadataUpdated implements updates to metadataMapCached via simple polling every
+// MetadataPollingService performs metadata-related services via simple polling every
 // interval duration.
-func KeepMetadataUpdated(client *http.Client, logger shared.Logger, interval time.Duration) {
-	logger.Infof("Metadata update via polling started")
+func MetadataPollingService(ctx context.Context, logger shared.Logger, interval time.Duration) {
+	logger.Infof("Metadata polling service started")
```

Comment:
Super-nit: this log is technically inaccurate as the service hasn't started yet, so it could cause confusion if something goes wrong below. I.e. you'd get the log lines:

```
Metadata polling service started
Unable to get GitHub client: foo bar
```

So if you read just the log lines, you might think the service has started :). Probably worth making it near-future tense (`starting`).